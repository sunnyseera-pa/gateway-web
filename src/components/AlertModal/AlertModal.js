/** @jsx jsx */
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import { Modal } from 'react-bootstrap';
import * as styles from './AlertModal.styles';

const AlertModal = ({ header, body, footer, ...outerProps }) => {
    return (
        <Modal {...outerProps} aria-labelledby='contained-modal-title-vcenter' centered>
            <Modal.Header>
                <Modal.Title id='contained-modal-title-vcenter' data-testid='modal-header'>
                    {header}
                </Modal.Title>
            </Modal.Header>
            <Modal.Body css={styles.modalBody} className='gray800-14' data-testid='modal-body'>
                {body}
            </Modal.Body>
            <Modal.Footer css={styles.modalFooter} data-testid='modal-footer'>
                {footer}
            </Modal.Footer>
        </Modal>
    );
};

AlertModal.propTypes = {
    onHide: PropTypes.func,
    show: PropTypes.bool,
    variant: PropTypes.oneOf(['error', 'success', 'info', 'warning']),
    header: PropTypes.node,
    body: PropTypes.node,
    footer: PropTypes.node,
};

AlertModal.defaultProps = {
    show: false,
};

export default AlertModal;
