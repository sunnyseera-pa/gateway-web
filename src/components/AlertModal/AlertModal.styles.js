import { css } from '@emotion/react';

export const modalBody = css`
    text-align: center;
`;

export const modalFooter = css`
    border-top: none !important;
    padding-left: 32px !important;
    padding-right: 32px !important;
`;
