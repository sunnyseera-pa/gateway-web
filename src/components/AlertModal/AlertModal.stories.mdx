import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { Button } from 'react-bootstrap';
import AlertModal from './AlertModal';

<Meta
	title='Core/AlertModal'
	component={AlertModal}
	args={{
		header: 'Oops! Something went wrong!',
		body: 'This issue has been automatically reported to our team!',
		variant: 'error',
	}}
/>

<Canvas>
	<Story name='Basic'>
		{props => {
			const [show, setShow] = React.useState(true);
			const handleClose = React.useCallback(() => {
				setShow(false);
			}, []);
			const handleOpen = React.useCallback(() => {
				setShow(true);
			}, []);
			return (
				<>
					{!show && <Button onClick={handleOpen}>Show modal</Button>}
					<AlertModal {...props} show={show} footer={<Button onClick={handleClose}>Close</Button>} />
				</>
			);
		}}
	</Story>
</Canvas>
