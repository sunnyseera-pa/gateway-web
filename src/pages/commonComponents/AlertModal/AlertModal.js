/** @jsx jsx */
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import { Modal } from 'react-bootstrap';
import * as styles from './AlertModal.styles';

const AlertModal = ({ header, body, footer, ...outerProps }) => {
	return (
		<Modal {...outerProps} aria-labelledby='contained-modal-title-vcenter' centered>
			<Modal.Header>
				<Modal.Title id='contained-modal-title-vcenter'>{header}</Modal.Title>
			</Modal.Header>
			<Modal.Body css={styles.modalBody} className='gray800-14'>
				{body}
			</Modal.Body>
			<Modal.Footer css={styles.modalFooter}>{footer}</Modal.Footer>
		</Modal>
	);
};

AlertModal.propTypes = {
	onHide: PropTypes.func,
	show: PropTypes.bool,
	variant: PropTypes.oneOf(['error', 'success', 'info', 'warning']),
	header: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
	body: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
	footer: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
};

AlertModal.defaultProps = {
	show: false,
};

export default AlertModal;
