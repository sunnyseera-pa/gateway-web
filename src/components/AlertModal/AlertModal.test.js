import { render } from '@testing-library/react';
import React from 'react';
import AlertModal from '.';

let wrapper;

const props = {
    show: true,
    header: 'This is the header',
    body: 'This is the body',
    footer: 'This is the footer',
};

let containerDiv;

describe('Given the AlertModal component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            containerDiv = createPortalContainer();

            wrapper = render(<AlertModal {...props} container={containerDiv} />);
        });

        afterAll(() => {
            removePortalContainer(containerDiv);
        });

        it('Then matches the previous snapshot', () => {
            expect(containerDiv).toMatchSnapshot();
        });

        it('Then has the correct header', () => {
            expect(wrapper.getByTestId('modal-header').textContent).toBe('This is the header');
        });

        it('Then has the correct body', () => {
            expect(wrapper.getByTestId('modal-body').textContent).toBe('This is the body');
        });

        it('Then has the correct footer', () => {
            expect(wrapper.getByTestId('modal-footer').textContent).toBe('This is the footer');
        });

        describe('And the modal is hidden', () => {
            it('Then has the correct output', () => {
                wrapper.rerender(<AlertModal {...props} show={false} />);

                expect(wrapper.container.querySelector('.modal')).toBeNull();
            });
        });
    });
});
