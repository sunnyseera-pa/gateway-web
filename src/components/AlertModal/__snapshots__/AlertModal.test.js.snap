// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AlertModal component When it is rendered Then matches the previous snapshot 1`] = `
<div
  class="modal-open"
  style=""
>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-labelledby="contained-modal-title-vcenter"
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
            data-testid="modal-header"
            id="contained-modal-title-vcenter"
          >
            This is the header
          </div>
        </div>
        <div
          class="gray800-14 css-1axlqcz-AlertModal modal-body"
          data-testid="modal-body"
        >
          This is the body
        </div>
        <div
          class="css-n34xqh-AlertModal modal-footer"
          data-testid="modal-footer"
        >
          This is the footer
        </div>
      </div>
    </div>
  </div>
</div>
`;
